atos/modules/MQTTBridge/src/mqttbridge.cpp

```diff
@@ -4,123 +4,94 @@
+
+  this->initialize();
 }
 
 /*!
- * \brief Yield MQTT client at intervals shorter then the MQTT_KEEPALIVE_INTERVAL_S to keep the connection open
- *        if no messages are published for a while.
+ * \brief initializeModule Initializes this module by starting the mqtt client
+ * with ros parameter settings.
  */
-void MqttBridge::yieldMqttClient()
-{
-	MQTTClient_yield();
+void MqttBridge::initialize() {
+  if (this->brokerIP.empty()) {
+    RCLCPP_INFO(this->get_logger(),
+                "No Broker IP provided in configuration. Shutting down...");
+    rclcpp::shutdown();
+  } else {
+    this->setupConnection();
+  }
 }
 
-void MqttBridge::setupConnection()
-{
-	// Add a random number to avoid conflicting client IDs
-	std::random_device rd;
-	std::uniform_int_distribution<int> dist(10000, 99999);
-	static const std::string client_id = pubClientId + std::to_string(dist(rd));
-
-	RCLCPP_INFO(this->get_logger(), "Setting up connection with clientID: %s, and broker IP: %s", client_id.c_str(), brokerIP.c_str());
-
-	MQTTClient mqttClient = MQTT::setupConnection(brokerIP.c_str(),
-												  client_id.c_str(),
-												  username.c_str(),
-												  password.c_str(),
-												  MQTT::clientType::publisher,
-												  "");
-
-	if (mqttClient == nullptr)
-	{
-		RCLCPP_ERROR(this->get_logger(), "Failed to initialize MQTT connection to broker, exiting...");
-		rclcpp::shutdown();
-	}
-	else
-	{
-		RCLCPP_DEBUG(this->get_logger(), "Successfully initialized MQTT connection to broker");
-		this->mqttClient = mqttClient;
-	}
+void MqttBridge::setupConnection() {
+  RCLCPP_INFO(this->get_logger(),
+              "Setting up connection with clientID: %s, and broker IP: %s",
+              clientId.c_str(), brokerIP.c_str());
+
+  mqttClientWrapper =
+      std::make_shared<MQTTClientWrapper>(brokerIP, port, username, password);
+  mqttClientWrapper->connect();
+
+  if (mqttClientWrapper == nullptr) {
+    RCLCPP_ERROR(this->get_logger(),
+                 "Failed to initialize MQTT connection to broker, exiting...");
+    rclcpp::shutdown();
+  } else {
+    RCLCPP_DEBUG(this->get_logger(),
+                 "Successfully initialized MQTT connection to broker");
+  }
 }
 
-void MqttBridge::onV2xMsg(const V2X::message_type::SharedPtr v2x_msg)
-{
-	if (MQTTClient_isConnected(this->mqttClient))
-	{
-		json payload = v2xToJson(v2x_msg);
-		const MQTT::Message mqttMsg = MQTT::Message(payload.dump(), topic.c_str());
-		bool retained = false;
-
-		try
-		{
-			RCLCPP_DEBUG(this->get_logger(), "Publishing MQTT v2x msg to broker %s", payload.dump().c_str());
-			MQTT::publishMessage(mqttMsg, this->mqttClient, retained);
-		}
-		catch (std::runtime_error&)
-		{
-			RCLCPP_ERROR(this->get_logger(), "Failed to publish MQTT message");
-		}
-	}
-	else
-	{
-		RCLCPP_ERROR(this->get_logger(), "Received v2x msg while the client is disconnected, dropping msg...");
-	}
+void MqttBridge::onV2xMsg(const V2X::message_type::SharedPtr v2x_msg) {
```

Comment:
```suggestion
void MqttBridge::onV2xMsg(const V2X::message_type::SharedPtr v2xMsg) {
```
Doesn't follow the convention ;)

Reply:
I agree and disagree. This doesn't follow normal cpp convention. However, this follows ROS convention, which is "biased" towards python convention to make use of ROS across languages more unified